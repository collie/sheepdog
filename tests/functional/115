#!/bin/bash

# Test fixed vnodes feature -- join

. ./common

_need_to_be_root

function setUpDevice
{
	_make_device 0 $((100*1024**2))
	_make_device 1 $((200*1024**2))
	_make_device 2 $((300*1024**2))
}

function setUpAutoNode
{
	SHEEP_OPTIONS='' _start_sheep 0
	SHEEP_OPTIONS='' _start_sheep 1
	SHEEP_OPTIONS='' _start_sheep 2
	_wait_for_sheep 3
}

function setUpAutoCluster
{
	setUpDevice
	setUpAutoNode
	_cluster_format -c 2
}

function setUpFixedNode
{
	SHEEP_OPTIONS='-V 100' _start_sheep 0
	SHEEP_OPTIONS='-V 200' _start_sheep 1
	SHEEP_OPTIONS='-V 300' _start_sheep 2
	_wait_for_sheep 3
}

function setUpFixedCluster
{
	setUpDevice
	setUpFixedNode
	_cluster_format -V -c 2
}

# Give me SHEEP_OPTIONS when you call testJoinSucceeded
function testJoinSucceeded
{
	"$1" # setUpFoobarCluster
	$DOG cluster info -v | _filter_cluster_info

	_make_device 3 $((400*1024**2))
	_start_sheep 3
	_wait_for_sheep 4
	for i in 0 1 2 3 ; do
		_wait_for_sheep_recovery "$i"
	done
	$DOG cluster info -v | _filter_cluster_info

	_cleanup
}

# Give me SHEEP_OPTIONS when you call testJoinFailed
function testJoinFailed
{
	"$1" # setUpFoobarCluster
	$DOG cluster info -v | _filter_cluster_info

	_make_device 3 $((400*1024**2))
	_start_sheep 3
	_wait_for_sheep_stop 3
	_wait_for_sheep 3
	for i in 0 1 2 ; do
		_wait_for_sheep_recovery "$i"
	done
	$DOG cluster info -v | _filter_cluster_info

	_cleanup
}

SHEEP_OPTIONS=''       testJoinSucceeded setUpAutoCluster
SHEEP_OPTIONS='-V 400' testJoinSucceeded setUpFixedCluster
SHEEP_OPTIONS=''       testJoinFailed    setUpFixedCluster
SHEEP_OPTIONS='-V 400' testJoinFailed    setUpAutoCluster
