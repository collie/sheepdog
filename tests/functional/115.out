QA output created by 115
using backend plain store
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: auto
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:63, 127.0.0.1:7001:128, 127.0.0.1:7002:193]
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: auto
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:50, 127.0.0.1:7001:102, 127.0.0.1:7002:154, 127.0.0.1:7003:206]
DATE      1 [127.0.0.1:7000:63, 127.0.0.1:7001:128, 127.0.0.1:7002:193]
using backend plain store
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: fixed
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300]
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: fixed
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300, 127.0.0.1:7003:400]
DATE      1 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300]
using backend plain store
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: fixed
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300]
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: fixed
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300]
DATE      2 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300, 127.0.0.1:7003:128]
DATE      1 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300]
using backend plain store
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: auto
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:63, 127.0.0.1:7001:128, 127.0.0.1:7002:193]
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: auto
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:63, 127.0.0.1:7001:128, 127.0.0.1:7002:193]
DATE      2 [127.0.0.1:7000:50, 127.0.0.1:7001:102, 127.0.0.1:7002:154, 127.0.0.1:7003:206]
DATE      1 [127.0.0.1:7000:63, 127.0.0.1:7001:128, 127.0.0.1:7002:193]
